### PR TITLE
Semver with bootstrapping

### PR DESCRIPTION
**Pull request by aaronpowell**
Here's my thoughts based on the comments I created in PR #39 on how to clean up the DLLs and other stuff

### CODE REVIEW

File: .bootstrapper/SemVerBootstrapper.ps1 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#requires -version 3.0
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+if (Get-Command dotnet -ErrorAction SilentlyContinue) {
+    dotnet publish -o .
```
Comment (felixfbecker) [inline .bootstrapper/SemVerBootstrapper.ps1:6]: doesn't this mean every user needs to have the dotnet SDK/CLI installed?

Comment (aaronpowell) [reply .bootstrapper/SemVerBootstrapper.ps1:6]: Yes, but the intent would be that it is only run on the build server or by someone who's cloned it.

**Comment by aaronpowell** [comment]
The other advantage of this approach is that it makes it easier to do local development, you don't need to run some step to do an install for local development/clone-based installs.

File: .bootstrapper/SemVerBootstrapper.ps1 (hunk 1 above)
Comment (felixfbecker) [reply .bootstrapper/SemVerBootstrapper.ps1:6]: Oh, got it. I find this kinda weird to be honest. I've never seen a software that had code in the source to check whether the dependencies are installed and if not does that. I would rather expect the readme or contribution guide to show the command to run to install dependencies, next to the command how to run tests. `Install-Module Pester` is another thing that one has to do to run the tests anyway.

**Comment by aaronpowell** [comment]
The other advantage of this approach is that it makes it easier to do local development, you don't need to run some step to do an install for local development/clone-based installs.

**Comment by felixfbecker** [comment]
I just think it's weird to do a `Test-Path` and all that logic in the release runtime code, on every import. That's like having an npm package where the index.js beings with something like
```js
if (!fs.existsSync(__dirname + '/node_modules')) {
  cp.execSync('npm install')
}
```

**Comment by aaronpowell** [comment]
Looks like we can achieve this is just as different processes.

**Pull request closed without merging**